diff --git a/modules/mailchimp_signup/mailchimp_signup.module b/modules/mailchimp_signup/mailchimp_signup.module
index 06b0d55..abd2216 100644
--- a/modules/mailchimp_signup/mailchimp_signup.module
+++ b/modules/mailchimp_signup/mailchimp_signup.module
@@ -92,7 +92,7 @@ function mailchimp_signup_block_view($delta = '') {
   $signup = mailchimp_signup_load($delta);
   if ($signup) {
     $block = array(
-      'subject' => $signup->title,
+      'subject' => t($signup->title),
       'content' => drupal_get_form('mailchimp_signup_subscribe_block_' . $signup->name . '_form', $signup, 'mailchimp_signup_block'),
     );
   }
@@ -244,7 +244,7 @@ function mailchimp_signup_forms($form_id, $args) {
 function mailchimp_signup_subscribe_form($form, &$form_state, $signup, $type) {
   $form['#attributes'] = array('class' => array('mailchimp-signup-subscribe-form'));
   $form['description'] = array(
-    '#markup' => $signup->settings['description'],
+    '#markup' => t($signup->settings['description']),
   );
   $form['mailchimp_lists'] = array('#tree' => TRUE);
   $lists = mailchimp_get_lists($signup->mc_lists);
@@ -294,13 +294,16 @@ function mailchimp_signup_subscribe_form($form, &$form_state, $signup, $type) {
   foreach ($signup->settings['mergefields'] as $tag => $mergevar) {
     if (!empty($mergevar)) {
       $form['mergevars'][$tag] = mailchimp_insert_drupal_form_tag($mergevar);
+      if (isset($form['mergevars'][$tag]['#title'])){
+        $form['mergevars'][$tag]['#title'] = t($form['mergevars'][$tag]['#title']);
+      }
     }
   }
 
   $form['submit'] = array(
     '#type' => 'submit',
     '#weight' => 10,
-    '#value' => $signup->settings['submit_button'],
+    '#value' => t($signup->settings['submit_button']),
   );
 
   return $form;
@@ -365,7 +368,7 @@ function mailchimp_signup_subscribe_form_submit($form, &$form_state) {
     }
   }
   if (count($successes) && strlen($signup->settings['confirmation_message'])) {
-    drupal_set_message($signup->settings['confirmation_message'], 'status');
+    drupal_set_message(t($signup->settings['confirmation_message']), 'status');
   }
   if (!empty($signup->settings['destination'])) {
     $form_state['redirect'] = $signup->settings['destination'];
